azure/functions/decorators/function_app.py

```diff
@@ -2774,7 +2775,13 @@ def __init__(self, app,
         on the request in order to invoke the function.
         """
         super().__init__(auth_level=http_auth_level)
-        self._add_http_app(AsgiMiddleware(app))
+        self.middleware = AsgiMiddleware(app)
+        self._add_http_app(self.middleware)
+        self.startup_task_done = False
+
+    def __del__(self):
+        if self.startup_task_done:
+            asyncio.run(self.middleware.notify_shutdown())
```

Comment:
curious why for shutdown, we dont raise runtimeerror if shutdown does not succeed (similar to startup)?